docs/apps/build-system.adoc

```diff
@@ -89,33 +89,54 @@ NOTE:  XP apps deployed via file (locally) are presented with a small blue icon
 
 === Development Mode
 
-Enonic XP supports a so-called Development mode.
-When running in this mode, XP will automatically source JavaScript controllers directly from your project folder.
+[WARNING]
+====
+As the name implies, development mode is intended only for development use and should _not_ be used in production. Doing so would cause both security and performance issues.
+====
 
-This may be convenient if you are developing with pure JavaScript and do not depend on build steps in your project.
+Enonic XP can be run in a special mode known as _development mode_. In this mode, XP will try to detect changes to your project source files, including JavaScript controllers and registered XML schemas. It does this by reading these files directly from their source locations instead of from the application `.jar` file.
 
-You may activate dev mode when starting XP as follows:
+Because of the way XP reads source files directly in development mode, it has a few limitations:
+
+- Files that require any sort of compilation (including transpilation) are not detected unless you run a build process on the side. For instance, if you use TypeScript or JavaScript features that XP doesn't support, these files won't be picked up by default. To have XP detect them, run a separate build process on the side, such as either a xref:build-system#_continuous_building[continuous build] or a webpack build.
++
+NOTE: XP uses the  Nashorn JavaScript engine for compiling JavaScript. For an overview over what JavaScript features Nashorn supports, see https://kangax.github.io/compat-table/es6/#nashorn1_8[this feature table].
++
+- XP will not pick up _new_ content types. If you add a new content type after your previous project deploy, XP will not be able to see it. However, changes to _existing_ content types _are_ picked up. XP will pick up new JavaScript controllers as long as it can run them without compilation (as per the previous point).
+
+If you need any of the above features, try using xref:build-system#_continuous_building[continuous build mode] instead.
+
+Development mode also disables some of XP's caching mechanisms. To make the development workflow as smooth as possible, XP tries to invalidate caches for your static assets. This is to prevent you from getting stale resources so that what you see in the browser is always as up to date as possible.
+
+Running XP in development mode may be convenient if your project does not rely on additional build steps such as JavaScript bundling. If you need to run arbitrary build tasks before deploying your project, check out xref:build-system#_continuous_building[the section on continuous builds].
```

Comment:
This is confusing... webpack recreates the bundle in build directory and new bundle sis picked up nicely.MAybe remove this paragraph alltogether?

Reply:
```suggestion
```